rpcq/_server.py

```diff
@@ -102,8 +167,12 @@ def exit_handler(self, f: Callable):
                     # empty_frame may either be:
                     # 1. a single null frame if the client is a REQ socket
                     # 2. an empty list (ie. no frames) if the client is a DEALER socket
-                    identity, *empty_frame, msg = done.result()
+                    identity, *empty_frame, msg, client_key = done.result()
                     request = from_msgpack(msg)
+                    try:
+                        request.params['client_key'] = client_key
+                    except Exception as e:
```

Comment:
When could this possibly get triggered? The incoming message is unpacked according to the server's local message spec, which will never go out of sync / always have a `client_key` field.

Reply:
Good catch! I wasn't sure if it would be better to have `client_key` be `Optional[str]` or default to "" in the request message - and I implemented it inconsistently here vs. there. What do you think?

Reply:
My preference is for `Optional[str]`; I like the ability to tell via `None` that there's no possible way a client could have come up with that value.

This reminds me: even if you aren't updating the Lisp client/server code, I'd prefer you to use the Lisp code generation tool for the python message spec. In `src/messages.lisp` you'll find an `|RPCRequest|` region where you can define this `client_key` field, & then in toplevel there's a python script (which wields some dockerized Lisp image) to update the corresponding python definition.

Reply:
Good suggestion, thanks! Done - added the field in lisp and ran the codegen. Field is now optional.

Reply:
Could we be more specific and catch `KeyError` (if that is indeed the expected error mode)?